applications/MappingApplication/custom_mappers/projection_3D_2D_mapper.h

```diff
@@ -0,0 +1,743 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "interpolative_mapper_base.h"
+#include "custom_mappers/nearest_neighbor_mapper.h"
+#include "custom_mappers/nearest_element_mapper.h"
+#include "custom_mappers/barycentric_mapper.h"
+#include "utilities/geometrical_projection_utilities.h"
+#include "utilities/parallel_utilities.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+/**
+ * @brief Type of meta mapper considered
+ */
+enum class MetaMapperType
+{
+    NEAREST_NEIGHBOR,
+    NEAREST_ELEMENT,
+    BARYCENTRIC
+};
+
+/**
+ * @brief Entity type mesh considered
+ */
+enum class EntityTypeMesh
+{
+    NONE,
+    CONDITIONS,
+    ELEMENTS
+};
+
+///@}
+///@name  Functions
+///@{
+
+/**
+ * @brief This method determines the partition where the model part has at least one entity
+ * @param rModelPart The input model part where determine the partition with at least one entity
+ * @return A partition with at least one partition
+ */
+int DeterminePartitionWithEntities(const ModelPart& rModelPart)
+{
+    const bool is_distributed = rModelPart.IsDistributed();
+    Geometry<Node<3>>::Pointer p_geometry = rModelPart.NumberOfElements() > 0 ? rModelPart.ElementsBegin()->pGetGeometry() : rModelPart.NumberOfConditions() > 0 ? rModelPart.ConditionsBegin()->pGetGeometry() : nullptr;
+
```

Comment:
I think an alternative way of doing this:
```c++
local_dimension = (p_geometry == nullptr) ? 0 : p_geometry->LocalDimension()
max_local_dimension = r_data_communicator.Max(local_dimension)
this_partition_is_2d = (max_local_dimension == 2);
```

Reply:
Let me see

Reply:
Done